src/actions/indexables/indexable-action.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace Yoast\WP\SEO\Actions\Indexables;
+
+use Yoast\WP\SEO\Helpers\Indexables_Page_Helper;
+use Yoast\WP\SEO\Helpers\Options_Helper;
+use Yoast\WP\SEO\Helpers\Post_Type_Helper;
+use Yoast\WP\SEO\Repositories\Indexable_Repository;
+
+/**
+ * Get action for indexables.
+ */
+class Indexable_Action {
+
+	/**
+	 * The indexable repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	private $indexable_repository;
+
+	/**
+	 * The post type helper.
+	 *
+	 * @var Post_Type_Helper
+	 */
+	private $post_type_helper;
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * Indexable_Action constructor.
+	 *
+	 * @param Indexable_Repository $indexable_repository The indexable repository.
+	 * @param Post_Type_Helper     $post_type_helper The post type helper.
+	 * @param Options_Helper       $options_helper The options helper.
+	 */
+	public function __construct(
+		Indexable_Repository $indexable_repository,
+		Post_Type_Helper $post_type_helper,
+		Options_Helper $options_helper
+	) {
+		$this->indexable_repository = $indexable_repository;
+		$this->post_type_helper     = $post_type_helper;
+		$this->options_helper       = $options_helper;
+	}
+
+	/**
+	 * Get public sub types.
+	 *
+	 * @return array The subtypes.
+	 */
+	protected function get_public_sub_types() {
+		$object_sub_types = \array_values(
+			\array_merge(
+				$this->post_type_helper->get_public_post_types(),
+				\get_taxonomies( [ 'public' => true ] )
+			)
+		);
+
+		$excluded_post_types = \apply_filters( 'wpseo_indexable_excluded_post_types', [ 'attachment' ] );
+		$object_sub_types    = \array_diff( $object_sub_types, $excluded_post_types );
+		return $object_sub_types;
+	}
+
+	/**
+	 * Gets the posts with the smallest readability scores.
+	 *
+	 * @param int $limit The maximum amount of results to return.
+	 *
+	 * @return array The posts with the smallest readability scores as an array.
+	 */
+	public function get_least_readable( $limit ) {
+		$ignore_list = empty( $this->options_helper->get( 'least_readability_ignore_list', [] ) ) ? [ -1 ] : $this->options_helper->get( 'least_readability_ignore_list', [] );
```

Comment:
Can we assign the result of the helper's getter in a var before we check it if it's empty so that we don't have to use the getter twice?

I know it's not part of the refactor specs, but it's a good chance if we did that now in all endpoints and update unit tests as well :)